prime-dotnet-webapi/Services/AuthorizedUserService.cs

```diff
@@ -0,0 +1,137 @@
+                .SingleOrDefaultAsync();
+        }
+
+        public async Task<AuthorizedUserViewModel> GetAuthorizedUserForUserIdAsync(Guid userId)
+        {
+            return await GetBaseAuthorizedUserQuery()
+                .Where(au => au.Party.UserId == userId)
+                .ProjectTo<AuthorizedUserViewModel>(_mapper.ConfigurationProvider)
+                .SingleOrDefaultAsync();
+        }
+
+        public async Task<int> CreateOrUpdateAuthorizedUserAsync(AuthorizedUserChangeModel changeModel, ClaimsPrincipal user)
+        {
+            var authorizedUser = await GetBaseAuthorizedUserQuery()
+                .SingleOrDefaultAsync(au => au.Party.UserId == user.GetPrimeUserId());
+
+            if (authorizedUser == null)
+            {
+                var party = new Party
+                {
+                    Addresses = new List<PartyAddress>()
+                };
+                _context.Parties.Add(party);
+
+                authorizedUser = new AuthorizedUser
+                {
+                    Party = party,
+                    Status = AccessStatusType.UnderReview
+                };
+                _context.AuthorizedUsers.Add(authorizedUser);
+            }
+
+            changeModel.UpdateAuthorizedUser(authorizedUser, user);
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                return InvalidId;
+            }
+
+            return authorizedUser.Id;
+        }
+
+        public async Task ActivateAuthorizedUser(int authorizedUserId)
+        {
+            var authorizedUser = await _context.AuthorizedUsers
+                .SingleOrDefaultAsync(au => au.Id == authorizedUserId);
+
+            authorizedUser.Status = AccessStatusType.Active;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task ApproveAuthorizedUser(int authorizedUserId)
+        {
+            var authorizedUser = await _context.AuthorizedUsers
+                .SingleOrDefaultAsync(au => au.Id == authorizedUserId);
+
+            authorizedUser.Status = AccessStatusType.Approved;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteAuthorizedUserAsync(int authorizedUserId)
+        {
+            var authorizedUser = await _context.AuthorizedUsers
+                .SingleOrDefaultAsync(au => au.Id == authorizedUserId);
+
+            if (authorizedUser == null)
+            {
+                return;
+            }
+
+            await _partyService.DeletePartyAsync(authorizedUser.Party.Id);
+        }
+
+        private IQueryable<AuthorizedUser> GetBaseAuthorizedUserQuery()
+        {
+            return _context.AuthorizedUsers
+                .Include(au => au.Party)
+                    .ThenInclude(p => p.Addresses)
+                        .ThenInclude(pa => pa.Address)
```

Comment:
The reason you weren't seeing the Addresses after projecting is because `VerifiedAddress`, `PhysicalAddress`, and `MailingAddress` are not marked with [Computed], so wont be translated into SQL. The `PartyAddress` table is never SELECTed into and so is null. you could avoid these includes by updating `Party`:

```C#
[NotMapped]
[Computed]
public VerifiedAddress VerifiedAddress
{
    get => Addresses
        .Select(a => a.Address)
        .OfType<VerifiedAddress>()
        .SingleOrDefault();
}
```

But this change ould probably cause regressions on other Parties, so its fine for now.

Reply:
@james-hollinger when you're back can you provide recap on automapper.  I have a bunch of scattered pieces on how this works, and had a peak at the docs, but would be good to go over.

Reply:
@james-hollinger reminder to revisit and find out if there are indeed regressions, and what it would take to update it throughout - @mtpultz